test/test_fetcher.py

```diff
@@ -48,11 +48,20 @@ def test_AaFetcher_format_datetime(aa_fetcher):
 
 def test_AaFetcher_constructs_url_correctly(dummy_pv, aa_fetcher):
     aa_fetcher._url = 'dummy-url'
+    aa_fetcher._parse_raw_data = mock.MagicMock()
     early = pytz.utc.localize(EARLY_DATE)
     late = pytz.utc.localize(LATE_DATE)
-    constructed = aa_fetcher._construct_url(dummy_pv, early, late)
-    expected = 'dummy-url?pv=dummy&from=2001-01-01T01:01:00Z&to=2010-02-03T04:05:00Z'
-    assert constructed == expected
+
+    with mock.patch('requests.get') as mock_get:
```

Comment:
Interesting that you decided to mock and call `get_values()` rather than calling `_construct_url()`.

Reply:
Sorry, I mentioned this bit in the pull request description rather than the commit message!

I was trying to test whether providing `request_params` results in the correct URL being sent. Testing `_construct_url()` doesn't really check for this, as it is not the public interface that it will be called from.

Reply:
I know it's a bit strange that I mocked out `_parse_raw_data()`, but that's because these tests are applied to an abstract class. I didn't want to make the pull request too big!

Reply:
Sorry, so you did.

I think testing the public API is probably better.